tests/integration_test/transport_security_test/transport_security_test.go

```diff
@@ -149,3 +148,80 @@ func TestServiceControlWithTLS(t *testing.T) {
 		}()
 	}
 }
+
+func TestHttpsClients(t *testing.T) {
+	args := []string{
+		"--service_config_id=test-config-id",
+		"--rollout_strategy=fixed",
+		"--server_ssl_path=../../env/testdata/",
+	}
+
+	s := env.NewTestEnv(comp.TestHttpsClients, platform.EchoSidecar)
+	defer s.TearDown()
+	s.AppendHttpRules([]*annotationspb.HttpRule{
+		{
+			Selector: "1.echo_api_endpoints_cloudesf_testing_cloud_goog.Simpleget",
+			Pattern: &annotationspb.HttpRule_Get{
+				Get: "/simpleget",
+			},
+		},
+	})
+
+	if err := s.Setup(args); err != nil {
+		t.Fatalf("fail to setup test env, %v", err)
+	}
+
+	testData := []struct {
+		desc         string
+		httpsVersion int
+		certPath     string
+		port         uint16
+		wantResp     string
+		wantError    error
+	}{
+		{
+			desc:         "Succcess for HTTP1 client with TLS",
+			httpsVersion: 1,
+			certPath:     platform.GetFilePath(platform.ServerCert),
+			wantResp:     `simple get message`,
+		},
+		{
+			desc:         "Succcess for HTTP2 client with TLS",
+			httpsVersion: 2,
+			certPath:     platform.GetFilePath(platform.ServerCert),
+			wantResp:     `simple get message`,
+		},
+		{
+			desc:         "Fail for HTTP1 client, with incorrect key and cert",
+			httpsVersion: 1,
+			certPath:     platform.GetFilePath(platform.ProxyCert),
+			wantError:    fmt.Errorf("x509: certificate signed by unknown authority"),
+		},
+		{
+			desc:         "Fail for HTTP2 client, with incorrect key and cert",
+			httpsVersion: 2,
+			certPath:     platform.GetFilePath(platform.ProxyCert),
+			wantError:    fmt.Errorf("x509: certificate signed by unknown authority"),
+		},
+	}
+
+	for _, tc := range testData {
+		func() {
+			var resp []byte
+			var err error
+
+			url := fmt.Sprintf("https://localhost:%v/simpleget?key=api-key", s.Ports().ListenerPort)
+			resp, err = client.DoHttpsGet(url, tc.httpsVersion, tc.certPath)
+			if tc.wantError == nil {
+				if err != nil {
+					t.Fatal(err)
```

Comment:
Be careful with your `Fatal`, it will prevent other tests in the loop from running.

Reply:
If a test case fatal, why do we care about other cases?

Reply:
Good point